[export] kill _process_constraints()

pianpwk opened the pull request:
The process for populating range_constraints follows separate methods for non-strict (`make_constraints`), and strict (`_process_constraints`). The strict method is somewhat more convoluted, and the analysis that Dynamo performs for strict is already present as part of the non-strict process in make_constraints (produce_guards(), running the export constraint solver).

This PR kills _process_constraints() and replaces calls with make_constraints, without duplicating the work that Dynamo already does.

avikchaudhuri reviewed (approved):
Accepting to unblock, but not a fan of the flag. See inline comment for a suggestion on how to remove it.

pianpwk commented on torch/_export/non_strict_utils.py:
```diff
@@ -174,104 +173,119 @@ def make_fake_inputs(nn_module, args, kwargs, dynamic_shapes):
+    num_lifted_inputs: int,
+    equalities_inputs: Optional[EqualityConstraint] = None,
+    original_signature: Optional[inspect.Signature] = None,
+    skip_produce_guards: Optional[bool] = False,
```
Ah yes this makes complete sense.

pianpwk committed:
[export] kill _process_constraints() (#123985)

Summary:
The process for populating range_constraints follows separate methods for non-strict (`make_constraints`), and strict (`_process_constraints`). The strict method is somewhat more convoluted, and the analysis that Dynamo performs for strict is already present as part of the non-strict process in make_constraints (produce_guards(), running the export constraint solver).

This PR kills _process_constraints() and replaces calls with make_constraints, without duplicating the work that Dynamo already does.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/123985

Reviewed By: avikchaudhuri

Differential Revision: D56086223

Pulled By: pianpwk
